cirq/ops/eigen_gate.py

```diff
@@ -227,3 +234,45 @@ def _is_parameterized_(self) -> bool:
     def _resolve_parameters_(self: TSelf, param_resolver) -> TSelf:
         return self._with_exponent(
                 exponent=param_resolver.value_of(self._exponent))
+
+
+def _lcm(vals: Iterable[int]) -> int:
+    t = 1
+    for r in vals:
+        t = t * r // fractions.gcd(t, r)
+    return t
+
+
+def _approximate_common_period(periods: List[float],
+                               initial_denom: int = 60,
+                               reject_atol: float = 1e-8) -> Optional[float]:
+    if not periods:
+        return None
+    if any(e == 0 for e in periods):
+        return None
+    approx_rational_periods = [
+        fractions.Fraction(int(np.round(p * initial_denom)), initial_denom)
+        for p in periods
+    ]
+    common = float(_common_rational_period(approx_rational_periods))
+
+    for p in periods:
+        if p != 0 and abs(p * np.round(common / p) - common) > reject_atol:
+            return None
+
+    return common
+
+
+def _common_rational_period(rational_periods: List[fractions.Fraction]
+                            ) -> fractions.Fraction:
+    """Finds the least common integer multiple of some fractions.
+
+    The solution is the smallest positive integer c such that there
+    exists integers n_k satisfying p_k * n_k = c for all k.
+    """
+    assert rational_periods, "no well-defined solution for an empty list"
+    common_denom = _lcm(p.denominator for p in rational_periods)
```

Comment:
can you use numpy's lcm?

Reply:
I wish I could but we support a version of numpy that doesn't have that function. Also in Python 2 there is no LCM function under the math module.